tron/mesos.py

```diff
@@ -311,5 +340,13 @@ def _process_event(self, event):
             log.warn('Unknown type of event: {}'.format(event))
 
     def stop(self):
-        self.runner.stop()
-        self.deferred.cancel()
+        if self.runner:
+            self.runner.stop()
+        if self.deferred:
+            self.deferred.cancel()
+        for task in self.tasks.values():
+            task.log.warning(
+                'Still running during Mesos shutdown, becoming unknown'
+            )
+            task.exited(None)
+        self.tasks = {}
```

Comment:
nit: is it better to exit one task and delete that one from self.task? task.exited might be stuck for any reason.

Reply:
Sure. You can't modify something while you're iterating through it, but I made a copy to iterate through.